src/compression.rs

```diff
@@ -1455,7 +1499,8 @@ pub fn compression(
         if !res.is_empty() {
             // rewrite with the invention
             let res: CompressionStepResult = res[0].clone();
-            rewritten = res.rewritten.clone();
+            train_rewritten = res.train_data.rewritten.clone();
+            test_rewritten = if let Some(d) = &res.test_data { Some(d.rewritten.clone()) } else { None };
```

Comment:
See comment for line 96 in `compress.rs`.

Reply:
same solution as other places i think